### PR TITLE
Rework confocal downsampling tests

### PR DESCRIPTION
**Pull request by rpauszek**
Why this PR?
Continuing from #572, here I've tackled all of the tests dealing with kymo downsampling. Same strategy as before

### CODE REVIEW

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Good stuff! Most of it looks good, but I did leave a few comments.

One thing that I am wondering about is the `kymo_h5_file` which is now different. It is involved in more tests than the ones moved here, so I'm curious.

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py [hunk 1]
```diff
@@ -67,3 +66,105 @@ def roundoff_kymo():
...
+    ds.attrs["Start time (ns)"] = np.int64(start)
+    ds.attrs["Stop time (ns)"] = np.int64(start + len(ref.infowave.data.data) * dt)
+
+    # Force channel that overlaps kymo; step from high to low force
+    # We want two lines of the kymo to have a force of 30, the other 10.
+    # Force starts 5 samples before the kymograph.
+    # A kymotrack line is 20 samples long, with a 50 sample dead time on either side.
+    # The pause before the third line starts after 245 samples.
+    force_data = np.full(len(ref.infowave.data.data), 30)
+    force_data[260:] = 10
```
[the remark this answers is not in the document]
Comment (rpauszek) [reply lumicks/pylake/tests/test_imaging_confocal/conftest.py:140]: In the old fixture we use the same kymo as our typical test one ("Kymo1") which is analogous to what I do here. Just using sane values for the timestamps, as I've been introducing throughout.

I noticed the regression test too, but my original thought was to have a separate fixture for that so we don't get anything funky interfering with other tests. So basically we'd have a "good" fixture and a "weird" fixture. Looking now more closely at that test, we actually introduce the "weirdness" in the test itself (monkeypatch in a smaller stop time), so the changes made here should still work fine for that one. 

That being said, I agree it'd be nice to have some very different values in the deadtime so we'd easily spot off-by-one errors, so I'll add that part

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py [hunk 2]
```diff
@@ -67,3 +66,105 @@ def roundoff_kymo():
...
+        dt=int(1e9),
+        samples_per_pixel=5,
+        line_padding=2,
+    )
+
+    return kymo, ref
+
+
+@pytest.fixture(scope="module")
+def downsampled_results():
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/tests/test_imaging_confocal/conftest.py:73]: Fair enough!

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/tests/test_imaging_confocal/conftest.py]: Alright. Sounds reasonable to me!

**Review by JoepVanlier — APPROVED** [review]
:shipit:

**Pull request merged**